stdlib/mutex.ml

```diff
@@ -25,6 +25,5 @@ let[@inline] protect m f =
   | x ->
     unlock m; x
   | exception e ->
-    let bt = Printexc.get_raw_backtrace () in
     unlock m;
-    Printexc.raise_with_backtrace e bt
+    raise e
```

Comment:
Could you call `Fun.protect` instead of hand-rolling here? Then you would get the right behavior for backtraces (same as you implemented by hand), and the agreed-upon consensus if `unlock` itself fails.

Reply:
I'd love to, but I think `Fun` must depend on `Printexc` which depends on `Mutex`, so my hands are ~locked~ tied.

I had to change this for it to compile at all.